File: Day 03 - 21.03.2019/TypeScript/HomeActionFacade.ts
import {HomeAction} from "./HomeAction";
export class HomeActionFacade extends HomeAction{

	private constructor(){ super()}

    private static instance:HomeActionFacade=new HomeActionFacade();

    public static getInstance():HomeActionFacade{
        return HomeActionFacade.instance;
    }
	
	public Sleep(){  
		
		console.log("\n\n---------------------------------------")
		console.log("You have selected Sleep mode")
		console.log("---------------------------------------")
		console.log(this.CloseTheLight(true))
		console.log(this.TeethBrushing(true))
		console.log(this.CloseTheDoor(false))
		console.log("---------------------------------------\n\n")
	}

	public ExitTheHouse(){
		console.log("---------------------------------------")
		console.log("You have selected Exit mode from home")
		console.log("---------------------------------------")
		console.log(this.CloseTheDoor(false))
		console.log(this.CloseTheAir(true))
		console.log(this.CloseTheLight(false))
		console.log(this.OrganizationToExit(true))
		console.log("---------------------------------------\n\n")
	}
    


}

File: Day 03 - 21.03.2019/Python/HomeActionFacade.py
from HomeAction import*

class HomeActionFacade():
	_single=None 
	_instance = True
	def __init__( self ): 
		try:         
			HomeActionFacade._single = HomeAction()
		except Exception:
			HomeActionFacade._single = HomeActionFacade._single

		if HomeActionFacade._instance==True: 
			HomeActionFacade._instance = self 
		else:
			raise Exception(HomeActionFacade._instance)

	def Sleep(self):  

		print("\n\n---------------------------------------")
		print("You have selected Sleep mode")
		print("---------------------------------------")
		print(HomeActionFacade._single.CloseTheLight(False))
		print(HomeActionFacade._single.TeethBrushing(True))
		print(HomeActionFacade._single.CloseTheDoor(False))
		print("---------------------------------------\n\n")


	def ExitTheHouse(self):
		print("---------------------------------------")
		print("You have selected Exit mode from home")
		print("---------------------------------------")
		print(HomeActionFacade._single.CloseTheDoor(False))
		print(HomeActionFacade._single.CloseTheAir(True))
		print(HomeActionFacade._single.CloseTheLight(False))
		print(HomeActionFacade._single.OrganizationToExit(True))
		print("---------------------------------------\n\n")
    




